src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,265 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+  VNode
+} from "@stencil/core";
+import { getElementProp } from "../../utils/dom";
+import { TEXT } from "./resources";
+
+@Component({
+  tag: "calcite-inline-editable",
+  scoped: true,
+  styleUrl: "calcite-inline-editable.scss"
+})
+export class CalciteInlineEditable {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el!: HTMLCalciteInlineEditableElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Props
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify whether the wrapped input element is editable, defaults to false */
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  /** specify whether the confirm button should display a loading state, defaults to false */
+  @Prop({ mutable: true, reflect: true }) loading = false;
+
+  /** specify whether save/cancel controls should be displayed when editingEnabled is true, defaults to false */
+  @Prop({ reflect: true }) hasControls = false;
```

Comment:
maybe rename to just `controls`?

Reply:
I rather like the more descriptive adjective form. In my opinion, it makes the code easier to read.

Reply:
I'm just thinking about consistency with other HTML elements like the `video` element which just has `controls`. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video

Reply:
I'm ok with `hasControls` too but we should probably get someone else's opinion :)

Reply:
I usually prefer single word props so that when people are using it in HTML they don't have to type the dash, so `<calcite-inline-editable controls>` would be nicer than `<calcite-inline-editable has-controls>`

Reply:
I prefer the brevity of `controls` and it matches HTML spec in @driskull's example